### PR TITLE
[Refinement] cleanup ota_client_service.py, implement test file

### PR DESCRIPTION
**Pull request by Bodong-Yang**
## Description

This PR introduces some cleanup and replaces the old, duplicated codes in `ota_client_service.py` with 2 new functions, each create otaclient server instance and launch it. 
Now otaclient grpc server's address and port configurations are obtained directly in `ota_client_service`, not via `OtaClientStub`.
This PR also adds missing test files for `ota_client_service`. 

#### NOTE
This PR is split from #212 , as part of not directly related changes.

#### Motivation
About otaclient grpc server configuration, basically each module should only know what they need, `ota_client_stub` should only focus on API handling, so it doesn't need to know the server configuration, neither expose method to provide such information. Instead, the `ota_client_service` module should take the responsibility of loading such configuration.

For `ECUInfo`, in #212 , the new refactored `ota_client_stub.OTAClientServiceStub(renamed from OtaClientStub)` doesn't load ecu_info by itself and read global server_cfg, instead taking `ECUInfo` instance and server_cfg instance as init param, this aligns dependency injection principle and reduces coupling in some extend. 

Newly added `create_otaclient_grpc_server` works like a factory function, to load required information, and construct otaclient grpc server instance.

## Check list

- [x] test file(s) that covers the change(s) is implemented.
- [x] local test is passed. 

## Changes

In `ota_client_service.py`:
1. add `create_otaclient_grpc_server` function, which inits `OtaClientStub` and constructs otaclient grpc server instance,
2. add `launch_otaclient_grpc_server`, a one-call helper function to create and launch otaclient grpc server,
3. implement test files.

## Behavior changes

Does this PR introduce behavior change(s)?

- [x] Yes, internal behaivor (will not impact user experience).

### Previous behavior

In `launch_otaclient_grpc_server`, otaclient grpc server's listen address config is queried from `OtaClientStub`(and `OtaClientStub` obtains this config by loading ecu_info and read server_cfg).

### Behavior with this PR

In `ota_client_service`, in `create_otaclient_grpc_server`, now grpc server listen address/port configs are directly queried from loading `ecu_info` and `server_cfg`. 

## Breaking change

Does this PR introduce breaking change?
- [x] No.

### CODE REVIEW

**Comment by obi-t4** [comment]
> Previous behavior
In launch_otaclient_grpc_server, otaclient grpc server's listen address/listen port configs are queried from OtaClientStub(and OtaClientStub obtains these configs by loading ecu_info and read server_cfg).

The port is not queried from OtaClientStub, so modify the PR description.

**Review by obi-t4 — COMMENTED** [review]
I left a comment but basically it is OK.

File: tests/test_ota_client_service.py [hunk 1]
```diff
@@ -0,0 +1,118 @@
+# Copyright 2022 TIER IV, INC. All rights reserved.
```
[the remark this answers is not in the document]
Comment (Bodong-Yang) [reply tests/test_ota_client_service.py:1]: This will be handled in another chore PR I think? (all files should be updated)

**Comment by Bodong-Yang** [comment]
About copyright message, I will prepare another chore PR to update all other files's copyright message.

**Comment by Bodong-Yang** [comment]
> > Previous behavior
> > In launch_otaclient_grpc_server, otaclient grpc server's listen address/listen port configs are queried from OtaClientStub(and OtaClientStub obtains these configs by loading ecu_info and read server_cfg).
> 
> The port is not queried from OtaClientStub, so modify the PR description.

Updated.

**Review by obi-t4 — APPROVED** [review]
LGTM :+1:

**Pull request merged**